spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestSparkWriteConfDistributionMode.java

```diff
@@ -0,0 +1,169 @@
+    Assert.assertEquals(DistributionMode.HASH, writeConf.distributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithTableProperties() {
+    sql(
+        "CREATE TABLE %s (id BIGINT, data STRING, date DATE, ts TIMESTAMP) "
+            + "USING iceberg "
+            + "PARTITIONED BY (date, days(ts))",
+        tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    table.updateProperties().set(WRITE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_HASH).commit();
+
+    SparkWriteConf writeConf = new SparkWriteConf(spark, table, ImmutableMap.of());
+    Assert.assertEquals(DistributionMode.HASH, writeConf.distributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithTblPropAndSessionConfig() {
+    sql(
+        "CREATE TABLE %s (id BIGINT, data STRING, date DATE, ts TIMESTAMP) "
+            + "USING iceberg "
+            + "PARTITIONED BY (date, days(ts))",
+        tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    table.updateProperties().set(WRITE_DISTRIBUTION_MODE, WRITE_DISTRIBUTION_MODE_RANGE).commit();
+    spark
+        .conf()
+        .set(SparkSQLProperties.SESSION_WRITE_DISTRIBUTION_MODE, DistributionMode.HASH.modeName());
+
+    SparkWriteConf writeConf = new SparkWriteConf(spark, table, ImmutableMap.of());
+    // session config overwrite the table properties
+    Assert.assertEquals(DistributionMode.HASH, writeConf.distributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithWriteOptionAndSessionConfig() {
+    sql(
+        "CREATE TABLE %s (id BIGINT, data STRING, date DATE, ts TIMESTAMP) "
+            + "USING iceberg "
+            + "PARTITIONED BY (date, days(ts))",
+        tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    Map<String, String> writeOptions =
+        ImmutableMap.of(SparkWriteOptions.DISTRIBUTION_MODE, DistributionMode.RANGE.modeName());
+    spark
+        .conf()
+        .set(SparkSQLProperties.SESSION_WRITE_DISTRIBUTION_MODE, DistributionMode.HASH.modeName());
+
+    SparkWriteConf writeConf = new SparkWriteConf(spark, table, writeOptions);
+    // write options overwrite the session config
+    Assert.assertEquals(DistributionMode.RANGE, writeConf.distributionMode());
+  }
+
+  @Test
+  public void testSparkWriteConfDistributionModeWithEverything() {
+    sql(
+        "CREATE TABLE %s (id BIGINT, data STRING, date DATE, ts TIMESTAMP) "
+            + "USING iceberg "
+            + "PARTITIONED BY (date, days(ts))",
+        tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    Map<String, String> writeOptions =
+        ImmutableMap.of(SparkWriteOptions.DISTRIBUTION_MODE, DistributionMode.RANGE.modeName());
+    spark
+        .conf()
+        .set(SparkSQLProperties.SESSION_WRITE_DISTRIBUTION_MODE, DistributionMode.HASH.modeName());
```

Comment:
Nit could we have a newline after line 162 before the table.updateProperties(), makes it easier to read IMO.